File: front/app/types/feed.ts
/**
 * 订阅源相关类型定义
 */

/**
 * 订阅源数据模型
 */
export interface RssFeed {
  id: string
  title: string
  description: string
  url: string
  category: string
  icon?: string
  color?: string
  lastUpdated: string
  articleCount: number
  unreadCount?: number
  maxArticles?: number
  refreshInterval?: number
  refreshStatus?: 'idle' | 'refreshing' | 'success' | 'error'
  refreshError?: string
  lastRefreshAt?: string
  aiSummaryEnabled?: boolean
}

/**
 * 订阅源创建数据
 */
export interface CreateFeedData {
  url: string
  category_id?: number
  title?: string
  description?: string
  icon?: string
  color?: string
}

/**
 * 订阅源更新数据
 */
export interface UpdateFeedData {
  url?: string
  category_id?: number
  title?: string
  description?: string
  icon?: string
  color?: string
  max_articles?: number
  refresh_interval?: number
  ai_summary_enabled?: boolean
}

/**
 * RSS 响应数据
 */
export interface FeedResponse {
  status: string
  feed: {
    title: string
    description: string
    image?: string
  }
  items: FeedItem[]
}

/**
 * RSS 文章条目
 */
export interface FeedItem {
  title: string
  link: string
  pubDate: string
  description?: string
  content?: string
  author?: string
  thumbnail?: string
  enclosure?: {
    link: string
  }
}

File: backend-go/internal/models/feed.go
package models

import (
	"time"
)

type Feed struct {
	ID               uint       `gorm:"primaryKey" json:"id"`
	Title            string     `gorm:"size:200;not null" json:"title"`
	Description      string     `gorm:"type:text" json:"description"`
	URL              string     `gorm:"size:500;unique;not null" json:"url"`
	CategoryID       *uint      `gorm:"index" json:"category_id"`
	Icon             string     `gorm:"size:50;default:rss" json:"icon"`
	Color            string     `gorm:"size:20;default:#8b5cf6" json:"color"`
	LastUpdated      *time.Time `json:"last_updated"`
	CreatedAt        time.Time  `json:"created_at"`
	MaxArticles      int        `gorm:"default:100" json:"max_articles"`
	RefreshInterval  int        `gorm:"default:60" json:"refresh_interval"` // minutes
	RefreshStatus    string     `gorm:"size:20;default:idle" json:"refresh_status"`
	RefreshError     string     `gorm:"type:text" json:"refresh_error"`
	LastRefreshAt    *time.Time `json:"last_refresh_at"`
	AISummaryEnabled bool       `gorm:"default:true" json:"ai_summary_enabled"`
	Articles         []Article  `gorm:"foreignKey:FeedID;constraint:OnDelete:CASCADE" json:"articles,omitempty"`
}

func (f *Feed) ToDict(includeStats bool) map[string]interface{} {
	data := map[string]interface{}{
		"id":                 f.ID,
		"title":              f.Title,
		"description":        f.Description,
		"url":                f.URL,
		"category_id":        f.CategoryID,
		"icon":               f.Icon,
		"color":              f.Color,
		"last_updated":       FormatDatetimeCSTPtr(f.LastUpdated),
		"created_at":         FormatDatetimeCST(f.CreatedAt),
		"max_articles":       f.MaxArticles,
		"refresh_interval":   f.RefreshInterval,
		"refresh_status":     f.RefreshStatus,
		"refresh_error":      f.RefreshError,
		"last_refresh_at":    FormatDatetimeCSTPtr(f.LastRefreshAt),
		"ai_summary_enabled": f.AISummaryEnabled,
	}

	if includeStats {
		articleCount := len(f.Articles)
		unreadCount := 0
		for _, a := range f.Articles {
			if !a.Read {
				unreadCount++
			}
		}
		data["article_count"] = articleCount
		data["unread_count"] = unreadCount
	}

	return data
}
